### PR TITLE
feat(module:anchor): support standalone component

### PR DESCRIPTION
**Pull request by evgeniyefimov**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Transfromed Anchor module to standalone simillar to https://github.com/NG-ZORRO/ng-zorro-antd/pull/8037

Seems `BidiModule` is unused. Didn't find any usage of this module in the anchor's module. Removed.

Imported only required directives from `@angular/common` instead of whole `CommonModule`.

### CODE REVIEW

**Review by HyperLife1119 — DISMISSED** [review]
I don't think anchor should be a standalone component, because it usually requires the combination of multiple components (nz-anchor and nz-link) to be used. Anchor cannot be used alone.

All functions that require the combination of ***multiple*** components to function properly should not be standalone components. Instead, these components should be bundled into ***one*** feature through NgModule.

Based on this point of view, it is not suitable to be converted into a standalone component.

WDYT?

**Comment by HyperLife1119** [comment]
I don't think anchor should be a standalone component, because it usually requires the combination of multiple components (nz-anchor and nz-link) to be used. Anchor cannot be used alone. Based on this point of view, it is not suitable to be converted into a standalone component. WDYT?

**Comment by evgeniyefimov** [comment]
> I don't think anchor should be a standalone component, because it usually requires the combination of multiple components (nz-anchor and nz-link) to be used. Anchor cannot be used alone.
> 
> All functions that require the combination of _**multiple**_ components to function properly should not be standalone components. Instead, these components should be bundled into _**one**_ feature through NgModule.
> 
> Based on this point of view, it is not suitable to be converted into a standalone component.
> 
> WDYT?

Module is still there, so nz-anchor and nz-link bundled together when module is imported. Also docs mention module import. 

From user perspective nothing will be changed.

From developers point of view it will be easier to understand what dependencies are required in components and import only the necessary dependencies, thereby reducing the size of the bundle.

**Comment by HyperLife1119** [comment]
I have changed my mind and you are right.  We should not restrict whether users should import ng modules or standalone components, this should be left to the user's choice. 👍

**Review by HyperLife1119 — APPROVED** [review]
LGTM

File: components/anchor/anchor-link.component.ts [hunk 1]
```diff
@@ -26,6 +27,8 @@ import { NzAnchorComponent } from './anchor.component';
   selector: 'nz-link',
   exportAs: 'nzLink',
   preserveWhitespaces: false,
+  standalone: true,
+  imports: [NgIf, PlatformModule],
```
Comment (HyperLife1119) [inline components/anchor/anchor-link.component.ts:31]: I discovered that PlatformModule is actually an empty module, let's delete it safely :)

**Pull request merged**